1-development/list_posts_by_rating/src/main/java/com/mixfar/posts/controllers/PostController.java

```diff
@@ -0,0 +1,111 @@
+package com.mixfar.posts.controllers;
+
+import com.mixfar.posts.model.Post;
+import com.mixfar.posts.service.PostService;
+import com.mixfar.posts.utils.Utils;
+import com.mixfar.posts.utils.enums.ErrorCode;
+import com.mixfar.posts.utils.errors.ErrorMessage;
+import com.mixfar.posts.utils.errors.FieldErrorMessage;
+import com.mixfar.posts.utils.response.DataResponse;
+import com.mixfar.posts.utils.response.ErrorResponse;
+import com.mixfar.posts.utils.response.Response;
+import com.mixfar.posts.validations.PostValidation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+public class PostController {
+
+    @Autowired
+    private PostService postService;
+
+    @RequestMapping(value = "/create", method = RequestMethod.POST)
+    public ResponseEntity<Response> createPost(@RequestBody Post post){
+        post.setPostId(null);
+        List<FieldErrorMessage> errors = new PostValidation(postService).validCreate(post);
+        if (errors != null)
+            return new ResponseEntity<>(new ErrorResponse<>(ErrorCode.INVALID_PARAMETERS, errors), HttpStatus.BAD_REQUEST);
```

Comment:
Why do you return an object that contains a ```status code```, and another parameter with the same ```status code```?

Reply:
The object ErrorResponse have a `status code` as a parameter because is to send that `status code` on the body of the response